server/routes.go

```diff
@@ -187,14 +187,38 @@ func (s *RESTServer) Run() {
 		}()
 	}
 	log.Println("Listening on", s.PortNumber)
-	http.ListenAndServe(":"+s.PortNumber, s.addRoutes())
+
+	h := httpdown.HTTP{}
+	s.server, err = h.ListenAndServe(&http.Server{
+		Addr:    ":" + s.PortNumber,
+		Handler: s.addRoutes(),
+	})
+	if err != nil {
+		log.Println(err)
+		return err
+	}
+	return s.server.Wait()
+}
+
+// Stop will stop the server and return when all the server goroutines have
+// exited and the socked closed.
+func (s *RESTServer) Stop() error {
+	// first shutdown the transaction writing.
+	// We don't stop the fixity process. Should we?
```

Comment:
Eh. I don't know. It is a read only process which will stop when the process exits...
To stop it I'll need to add a waitgroup and a signal channel for the fixity goroutines. And, like transactions, it may take a while if it is checksuming a large multi GB item.